shapeshift/web
# feat: useFoxyBalances multi-account

## Description

Adds multi-account support to `useFoxyBalances`.

If an explicit `accountNumber` is not provided we now get the balances for all available accounts.

Key interface change for the `getFoxyBalances` API:

```diff
 type GetFoxyBalancesInput = {
-  accountId: AccountId
+  accountIds: AccountId[]
   foxyApr: string
 }
```

## Notice


## Pull Request Type


## Issue (if applicable)

Closes https://github.com/shapeshift/web/issues/3058

## Risk

Small as isolated to foxy balances and opportunities.

## Testing

1. Enable multi-account
2. Add another Ethereum account with FOX/FOXy

Consumers of `useFoxyBalances` should display aggregates across all accounts, e.g.:

- Fox Page
- DeFi page (both Overview and Earn)

### Engineering

☝️ 

### Operations

☝️ 

Staking, unstaking, claiming etc should still work as expected, particularly for opportunities related to FOX/FOXy.

## Screenshots (if applicable)

N/A
